src/Application/Lockers/Commands/RemoveLocker/RemoveLockerCommand.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Application.Common.Interfaces;
+using Application.Common.Models.Dtos.Physical;
+using AutoMapper;
+using Domain.Entities.Physical;
+using Domain.Exceptions;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace Application.Lockers.Commands.RemoveLocker;
+
+public record RemoveLockerCommand : IRequest<LockerDto>
+{
+    public Guid LockerId { get; init; }
+    public Guid RoomId { get; init; }
+}
+
+public class RemoveLockerCommandHandler : IRequestHandler<RemoveLockerCommand, LockerDto>
+{
+    private readonly IApplicationDbContext _context;
+    private readonly IMapper _mapper;
+
+    public RemoveLockerCommandHandler(IApplicationDbContext context, IMapper mapper)
+    {
+        _context = context;
+        _mapper = mapper;
+    }
+
+    public async Task<LockerDto> Handle(RemoveLockerCommand request, CancellationToken cancellationToken)
+    {
+        var locker = await _context.Lockers.Include(x => x.Room).FirstOrDefaultAsync(x => x.Id.Equals(request.LockerId) && x.IsAvailable == true, cancellationToken);
```

Comment:
add one more line with each method call, squeezing it into one method is terrible in many ways